core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1905,6 +1905,13 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
 
   @nowarn("cat=deprecation")
   private def validateValues(): Unit = {
+    val nodeIdValue = values.get(KafkaConfig.NodeIdProp).asInstanceOf[Int]
+    if (nodeIdValue >= 0) {
+      val brokerIdValue = values.get(KafkaConfig.BrokerIdProp).asInstanceOf[Int]
+      if (brokerIdValue != Defaults.BrokerId && brokerIdValue != nodeIdValue) {
+        throw new ConfigException(s"The values for broker.id ($brokerIdValue) and node.id ($nodeIdValue) must be the same if both are specified")
+      }
+    }
```

Comment:
Do you think we can put this check into below else block for KRaft-based metadata quorum case only?

Reply:
Good question.  It's legal to set node.id for the ZooKeeper case, so I think this inconsistency check is applicable in all cases and should be done here, at the top.